core/trino-main/src/main/java/io/trino/execution/scheduler/faulttolerant/EventDrivenFaultTolerantQueryScheduler.java

```diff
@@ -2331,8 +2332,7 @@ public void taskFinished(TaskId taskId, TaskStatus taskStatus)
                 // it is rare but possible to get empty spooling output stats for task which completed successfully.
```

Comment:
> it is rare but possible to get empty spooling output stats for task which completed successfully.

why it happens and can it be prevented instead?

Reply:
I think it is hard to prevent. 
The TaskStatus and TaskInfo are two separate channels. There will always be a possibility that we get TaskStatus saying that task FINISHED but cannot get final TaskInfo. Currently we decide to not fail task in this siutation - but recreate final TaskInfo locally from information we have at hand. And it does not work with output spooling stats. 
Alterantive would be to always kill task in such case - but I do not think it is better.